### DISCUSSION TITLE
nnn-newfile

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [jarun/nnn#1794] [Show and tell]
This is a `nnn` plugin that allows you to:

1. Call the plugin
2. Choose among a list of templates
3. Type a name
4. A  new file is created in the current folder by copying the template you chose into the name you chose

By default the folder `$HOME/Templates` is used for the templates.

It depends on `fzf` to select templates from a list.

https://github.com/raffaem/nnn-newfile

**Comment by jarun** [comment]
`nnn` has a keybind `n` for the same.

**Reply by raffaelemancuso** [reply]
No it hasn't, it's not template-based

**Reply by N-R-K** [reply]
> No it hasn't, it's not template-based

@raffaem Can you edit the OP to shorty describe what your plugin does, it's dependencies etc. For example, look at the other "show and tell" threads: https://github.com/jarun/nnn/discussions/categories/show-and-tell

I can reopen the discussion so that it gets more visibility.

**Reply by raffaelemancuso** [reply]
done!

**Comment by 62832** [comment]
The only thing I'd suggest is that you use `xdg-user-dir` to find the user's Templates folder instead of hard-coding against `~/Templates`, just in case it's set to something other than the default location. Otherwise, this is a pretty neat plugin.

**Reply by raffaelemancuso** [reply]
done, check it out!

**Reply by 62832** [reply]
Pretty good. Though, I would say the `~/Templates` directory could still be used as a fallback in case, say, `xdg-user-dir` isn't found as a command.

**Comment by raffaelemancuso** [comment]
> Pretty good. Though, I would say the `~/Templates` directory could still be used as a fallback in case, say, `xdg-user-dir` isn't found as a command.

Done. Check it again!
